### PR TITLE
Uprate python constraints to 3.7-3.10; remove six; uprate django version constraints.

### PR DESCRIPTION
**Pull request by dbenzel**
This PR establishes new version constraints for ```python```, ```django```, ```pytest```, adds python support up to 3.10, and removes the dependency on ```six```. Additionally, a ```SECRET_KEY``` is added to the tox.ini file to satisfy django 3.2's environment variable requirement for the same.

Major/minor constraints:
* python 3.7, 3.8, 3.9, 3.10
* django 2.2, 3.0, 3.1, 3.2
* pytest 6.2, 7.0, 7.1

@hearsaycorp/retain-and-grow @hearsaycorp/platform

### CODE REVIEW

File: setup.py
Comment (dbenzel) [reply setup.py:89]: Good callout. I was overruling myself because django has introduced more than a few breaking changes in minor versions, but that's no excuse to do the same here 👍🏻

File: tox.ini
Comment (cbueb) [inline tox.ini:7]: Looking into this more for other projects, I noticed that 3.6 isn't in any list of actively supported python versions: https://devguide.python.org/#status-of-python-branches

This may be something to consider dropping for the purposes of getting the build to pass.

Comment (dbenzel) [reply tox.ini:7]: This may eliminate unneeded complexity, true. It appeared to be working, with the remaining failures down to changes to django 3.2 requiring secret_key to make connections. I'm happy to remove 3.6 😄

Comment (dbenzel) [reply tox.ini]: Done

**Review by cbueb — COMMENTED** [review]
This seems good, but github says there's an error in the CODEOWNERS file.  I'll approve once that's fixed.

**Comment by dbenzel** [comment]
> This seems good, but github says there's an error in the CODEOWNERS file. I'll approve once that's fixed.

Thanks @cbueb, I have a question out to SRE to resolve, I believe it to be a write access issue for that github team.

**Review by ibiternas — APPROVED** [review]

**Review by epierce78 — APPROVED** [review]

**Pull request merged**
